src/MassTransit/Transports/BaseHost.cs

```diff
@@ -111,6 +126,8 @@ public async Task Stop(CancellationToken cancellationToken)
         {
             LogContext.Current = _hostConfiguration.LogContext;
 
+            await _riderCollection.Stop(cancellationToken).ConfigureAwait(false);
```

Comment:
So, this needs to stop the consumers on the rider, but not the producers, because receive endpoints may still be producing messages.